arelle/ModelValue.py

```diff
@@ -919,3 +922,34 @@ def __new__(cls, value: str) -> InvalidValue:
         return str.__new__(cls, value)
 
 INVALIDixVALUE = InvalidValue("(ixTransformValueError)")
+
+TypeSValue = Union[
+    bool,
+    float,
+    int,
+    InvalidValue,
+    None,
+    str,
+]
+TypeXValue = Union[
```

Comment:
Given the error thrown by tox, maybe we should make this a conditional depending on the Python version?